internal/util/paths/paths.go

```diff
@@ -79,17 +78,24 @@ func WriteString(path string, content string) error {
 	return err
 }
 
-// Expands leading tilde.
-func ExpandTilde(path string) (string, error) {
+// Expands home directory to absolute path.
+// If no expansion is required or possible, the string is returned unaltered.
+// If there is an error, an empty string is returned.
+func ExpandHomeDir(path string) (string, error) {
+
+	// Expand in case there are environment variables on the path
+	path = os.ExpandEnv(path)
 	usr, err := user.Current()
 	if err != nil {
```

Comment:
we only need the `usr` if `~` is present in the string. It would be perfectly valid to have an absolute path while running without a user (I don't know the use case of this, but it would be good future proofing)

Reply:
if `~` is present _and_ `err != nil`, then we `return "", err`

Reply:
I'm confused why we should return an empty string if `~` is present and the current user is resolvable? 

In this case, the user would just using their home directory for a notebook / some other path. Which is valid.

Reply:
@mcDevnagh ?